source/winVersion.py

```diff
@@ -81,32 +70,94 @@ def __ge__(self, other):
+WIN10_1909 = WinVersion(major=10, minor=0, build=18363, releaseName="Windows 10 1909")
+WIN10_2004 = WinVersion(major=10, minor=0, build=19041, releaseName="Windows 10 2004")
+WIN10_20H2 = WinVersion(major=10, minor=0, build=19042, releaseName="Windows 10 20H2")
+WIN10_21H1 = WinVersion(major=10, minor=0, build=19043, releaseName="Windows 10 21H1")
+
+# Records a mapping between Windows builds and release names.
+# These include build 10240 for Windows 10 1507 and releases with multiple release builds.
+# These are applicable to Windows 10 as they report the same system version (10.0).
+BUILDS_TO_RELEASE_NAMES = {
+	10240: "Windows 10 1507",
+	10586: "Windows 10 1511",
+	14393: "Windows 10 1607",
+	15063: "Windows 10 1703",
+	16299: "Windows 10 1709",
+	17134: "Windows 10 1803",
+	17763: "Windows 10 1809",
+	18362: "Windows 10 1903",
+	18363: "Windows 10 1909",
+	19041: "Windows 10 2004",
+	19042: "Windows 10 20H2",
+	19043: "Windows 10 21H1"
+}
+
+
+def getWindowsReleaseName(major: int = 0, minor: int = 0, build: int = 0) -> str:
+	"""Returns the public release name for a given Windows release based on major, minor, and build.
+	This is useful if release names are not defined as part of Windows version constants.
+	For example, 6.1 will return 'Windows 7'.
+	For Windows 10, feature update release name will be included.
+	On server systems, unless noted otherwise, client release names will be returned.
+	For example, 'Windows 10 1809' will be returned on Server 2019 systems.
+	"""
+	if (major, minor) == (6, 1):
+		return "Windows 7"
+	elif (major, minor) == (6, 2):
+		return "Windows 8"
+	elif (major, minor) == (6, 3):
+		return "Windows 8.1"
+	elif major == 10:
+		# From Version 1511 (build 10586), release Id/display version comes from Windows Registry.
+		# However there are builds with no release name (Version 1507/10240)
+		# or releases with different builds.
+		# Look these up first before asking Windows Registry.
+		if build in BUILDS_TO_RELEASE_NAMES:
+			return BUILDS_TO_RELEASE_NAMES[build]
+		# On Windows 10 1511 and later, cache the version in use on the system.
+		with winreg.OpenKey(
+			winreg.HKEY_LOCAL_MACHINE, r"Software\Microsoft\Windows NT\CurrentVersion"
+		) as currentVersion:
+			# Version 20H2 and later where a separate display version string is used.
+			try:
+				releaseId = winreg.QueryValueEx(currentVersion, "DisplayVersion")[0]
+			except OSError:
+				# Don't set anything if this is Windows 10 1507 or earlier.
+				try:
+					releaseId = winreg.QueryValueEx(currentVersion, "ReleaseID")[0]
+				except OSError:
+					releaseId = "unknown"
+		return f"Windows 10 {releaseId}"
+	else:
+		raise RuntimeError("Unknown Windows release")
 
 
 def getWinVer():
 	"""Returns a record of current Windows version NVDA is running on.
 	"""
 	winVer = sys.getwindowsversion()
+	try:
+		releaseName = getWindowsReleaseName(major=winVer.major, minor=winVer.minor, build=winVer.build)

```

Comment:
Try `_getRunningVersionNameFromReg` catch exception, then construct without a release name explicitly set, allowing `class WinVersion` to populate it via `_getWindowsReleaseName`.

Reply:
Yep - the new WinReg method will return an empty string for release Id, which is then processed inside the private release name method. This is actually a stepping stone toward #12585 once this PR is merged into master. Thanks.